block_my_feedback.php

```diff
@@ -59,35 +61,220 @@ public function get_content(): stdClass {
+                        $assess->url = new moodle_url('/mod/'. $mod->modname. '/view.php', ['id' => $cmid]);
+                        $assess->icon = course_summary_exporter::get_course_image($course);
+                        $marking[] = $assess;
+                    }
+                }
+            }
+        }
+
+        // Sort and return data.
+        if ($marking) {
+            usort($marking, function ($a, $b) {
+                return $a->unixtimestamp <=> $b->unixtimestamp;
+            });
+
+            return array_slice($marking, 0, 5);
+        }
+        return null;
+    }
+
+    /**
+     * Return mod data - due date & require marking.
+     *
+     * TODO - turnitin, quiz.
+     *
+     * @param stdClass $mod
+     * @param stdClass $assess
+     */
+    public function get_mod_data($mod, $assess) {
+        global $CFG;
+        // Mods have different feilds for due date, and require marking.
+        switch ($mod->modname) {
+            case 'assign':
+
+                // Check mod due date is relevant.
+                $duedate = self::duedate_in_range($mod->customdata['duedate']);
+                if (!$duedate) {
+                    return false;
+                }
+
+                // Add dates.
+                $assess->unixtimestamp = $duedate;
+                $assess->duedate = date('jS F y', $duedate);
+
+                // Require marking.
+                require_once($CFG->dirroot.'/mod/assign/locallib.php');
+                $context = context_module::instance($mod->id);
+                $assignment = new assign($context, $mod, $mod->course);
+                $assess->requiremarking = $assignment->count_submissions_need_grading();
+                if (!$assess->requiremarking) {
+                    return false;
+                }
+                $assess->markingurl = new moodle_url('/mod/'. $mod->modname. '/view.php',
+                    ['id' => $assess->cmid, 'action' => 'grader']
+                );
+                // Return template data.
+                return $assess;
+
+            // TODO - quiz - 'timeclose' ?.
+            case 'quiz':
+                return false;
+            // TODO - turnitin.
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * Return if course has started (startdate) and has not ended (enddate).
+     *
+     * @param stdClass $course
+     */
+    public function is_course_current(stdClass $course): bool {
+        // Start date.
+        if ($course->startdate > time()) {
+            return false; // Before the start date.
+        }
+
+        // End date.
+        if (isset($course->enddate)) {
+            if ($course->enddate == 0) {
+                return true; // Enddate is set to 0 when no end date, show course.
+            }
+            // Show course 3 months after end date to allow for UCL late summer assessments.
+            $cutoffdate = strtotime('+3 month');
+            if ($course->enddate > $cutoffdate) {
```

Comment:
This will prevent course being shown if its end date is more than 3 months from now.